EIPS/eip-747.md

```diff
@@ -0,0 +1,132 @@
+Web3 JavaScript wallet browsers may implement `wallet_watchAsset()` to allow any website to suggest a token for the user's wallet to track.
+
+## Motivation
+<!--The motivation is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. EIP submissions without sufficient motivation may be rejected outright.-->
+Today, one of the major uses of ethereum wallets is to acquire and track assets. Currently, each wallet either needs to pre-load a list of approved assets, or users need to be stepped through a tedious process of adding an asset for their given wallet.
+
+In the first case, wallets are burdened with both the security of managing this list, as well as the bandwidth of mass polling for known assets on their wallet.
+
+In the second case, the user experience is terrible.
+
+By leveraging a user's existing trust with websites they are learning about assets on, we are able to decentralize the responsibility of managing a user's list of known assets.
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+A new method is added to web3 browsers' ethereum providers:
+
+```javascript
+
+/**
+* @param  {Object}  opts - The options specifying the asset `type` and `options` specific for each of asset.
+* @returns  {Promise} success - Whether the user added the asset to their wallet.
+*/
+async  function  wallet_watchAsset (
+  opts
+) {
+/* Implementation would go here */
+}
+
+// Sample usage:
+web3.wallet.watchAsset({ type, options })
+```
+As there are several types of different assets, this method has to provide support for each of them in a separate way. If it doesn't, it should give a response according to that.
+
+In the case of assets of type `ERC20`, this method works as follows.
+
+```javascript
+web3.wallet.watchAsset({
+	type: 'ERC20',
+	options: { address, symbol, decimals [, image] }
+})
+```
+
+The `image` parameter should link to a web-standard image format (png, jpg) of a reasonable size or a `Base64` image. To establish a baseline, let's say no greater than 512x512 pixels, and no greater than 256kb. However, this can be a client-defined setting.
+
+An example of use in the first case would be.
+
+```javascript
+web3.wallet.watchAsset({
+	type: 'ERC20',
+	options: {
+		address,
+		symbol,
+		decimals,
+		image: 'linktoimage.jpg'
+	}
+})
+```
+Upon calling this request, the user should be prompted with the opportunity to add this token to their wallet:
+
+![ezgif com-video-to-gif](https://user-images.githubusercontent.com/12115171/44558730-2666e300-a71c-11e8-9098-e27ec99b88fb.gif)
```

Comment:
Damn, didn't notice these embedded gif links.

@danfinlay can you move them into `assets/eip-747`?

Reply:
Sorry about that: https://github.com/ethereum/EIPs/pull/2150